fdbclient/NativeAPI.actor.cpp

```diff
@@ -3554,9 +3630,19 @@ void Transaction::setOption( FDBTransactionOptions::Option option, Optional<Stri
 		    options.reportConflictingKeys = true;
 		    break;
 
-	    default:
+		case FDBTransactionOptions::TRANSACTION_IS_IDEMPOTENT:
+			validateOptionValue(value, false);
+			options.isIdempotent = true;
 			break;
-	}
+
+		case FDBTransactionOptions::RETRY_ON_UNKNOWN_RESULT:
+			validateOptionValue(value, false);
+			options.retryUnknownCommit = true;
+			break;
```

Comment:
I think we may need to flip this option on automatically for older API versions. We could instead flip on `TRANSACTION_IS_IDEMPOTENT` if we think that is better.